File: C#/백준/Silver/10773. 제로/제로.cs
using System;
using System.Collections.Generic;

namespace q{
    class w{
        static void Main(){
            int a = int.Parse(Console.ReadLine());
            int sum=0;
            Stack<int> stack = new Stack<int>();
            for (int i =0; i<a; i++){
                int b = int.Parse(Console.ReadLine());
                if (b==0){
                    stack.Pop();
                }
                else{
                    stack.Push(b);
                    }
                }
            foreach (int i in stack){
                sum+=i;
            }
            Console.WriteLine(sum);
        }
    }
}

File: Java/백준/Silver/10773. 제로/제로.java
import java.util.*;

public class Main{
    public static void main(String[] args){
        Scanner in = new Scanner(System.in);
        int a = in.nextInt();
        int sum=0;
        Stack<Integer> s = new Stack<>();
        for (int i=0; i<a; i++){
            int b = in.nextInt();
            if(b==0){
                s.pop();
            }
            else{
                s.push(b);
            }
        }
        for (int x:s){
            sum+=x;
        }
        System.out.println(sum);
    }
}